src/pages/playground/Playground.tsx

```diff
@@ -175,13 +197,61 @@ const Playground: React.FC<PlaygroundProps> = props => {
     }
   };
 
+  const uploadLogs = React.useCallback(() => {
+    const assessmentIDs = getLoggedAssessmentIds();
+    const assessmentLogs = getAssessmentLogs();
+    const playgroundLogs = getPlaygroundLogs();
+
+    if (assessmentLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(
+        assessmentIDs.assessmentId,
+        assessmentIDs.questionId,
+        assessmentLogs
+      );
+    }
+
+    if (playgroundLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(playgroundQuestionId, playgroundQuestionId, playgroundLogs);
+    }
+
+    resetPlaygroundLogging(props.sourceChapter, props.externalLibraryName, props.editorValue);
+  }, [props]);
+
+  const uploadPerHour = React.useCallback(() => {
```

Comment:
Knowing that 70kb of data was generated with short tests, do you think a 1 hour interval is suitable?

What happens when all logs are uploaded 1 hour after the start of the practical exam?

Reply:
In that case would it be better to set a limit to the # of logs that are stored? 

Then a time limit might not be necessary as such!